pkg/util/ip/addr_test.go

```diff
@@ -169,3 +170,80 @@ func TestFirstUsableIP(t *testing.T) {
 		g.Expect(FirstUsableIP(*cidr).String()).To(Equal(tc.expected))
 	}
 }
+
+func TestSplitNet(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	testcases := []struct {
+		cidr           string
+		expectedDouble string
+		expectedOther  string
+	}{
+		{
+			"10.0.0.0/24",
+			"10.0.0.0/25",
+			"10.0.0.128/25",
+		},
+		{
+			"10.0.0.128/28",
+			"10.0.0.128/29",
+			"10.0.0.136/29",
+		},
+		{
+			"172.30.0.0/16",
+			"172.30.0.0/17",
+			"172.30.128.0/17",
+		},
+		{
+			"10.0.128.0/23",
+			"10.0.128.0/24",
+			"10.0.129.0/24",
+		},
+	}
+
+	for _, tc := range testcases {
+		_, cidr, err := net.ParseCIDR(tc.cidr)
+		g.Expect(err).NotTo(HaveOccurred())
+		lower, upper := SplitNet(*cidr)
+		o, _ := lower.Mask.Size()
+		g.Expect(fmt.Sprintf("%s/%d", lower.IP.String(), o)).To(Equal(tc.expectedDouble))
+		o, _ = upper.Mask.Size()
+		g.Expect(fmt.Sprintf("%s/%d", upper.IP.String(), o)).To(Equal(tc.expectedOther))
+	}
+}
+
+func TestIterateIP4(t *testing.T) {
```

Comment:
you should really have at least one test that crosses a byte boundary. eg, `{ "10.0.0.0", -1, "9.255.255.255" }`

Reply:
Done.